### PR TITLE
chore(client): Typescript migration of __mocks__/gatsby.js

### PR DESCRIPTION
**Pull request by SadmanSyfe**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes part of #43778

### CODE REVIEW

**Review by Sembauke — DISMISSED** [review]

File: client/src/__mocks__/gatsby.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 /* eslint-disable no-unused-vars */
-const React = require('react');
+import { LinkProps } from '@reach/router';
```
Comment (ojeytonwilliams) [inline client/src/__mocks__/gatsby.ts:2]:
```suggestion
import { GatsbyLinkProps } from 'gatsby';
```
As I understand it, we're trying to replace gatsby's link, not reach router's.

File: client/src/__mocks__/gatsby.ts [hunk 2]
```diff
@@ -13,24 +14,14 @@ module.exports = {
...
     // these props are invalid for an `a` tag
-    ({
-      activeClassName,
-      activeStyle,
-      getProps,
-      innerRef,
-      partiallyActive,
-      ref,
-      replace,
-      to,
```
Comment (ojeytonwilliams) [inline client/src/__mocks__/gatsby.ts:24]: I believe these are all part of GatsbyLinkProps, so we should be discarding them.

File: client/src/__mocks__/gatsby.ts [hunk 3]
```diff
@@ -13,24 +14,14 @@ module.exports = {
...
+    ({ to, ...rest }: LinkProps<boolean | undefined>) =>
       React.createElement('a', {
         ...rest,
         href: to
       })
   ),
-  withPrefix: jest.fn().mockImplementation(path => {
-    const pathPrefix =
+  withPrefix: jest.fn().mockImplementation((path: string) => {
+    const pathPrefix: string =
```
Comment (ojeytonwilliams) [inline client/src/__mocks__/gatsby.ts:24]:
```suggestion
    const pathPrefix =
```
TS can infer this.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hey @SadmanSyfe, thanks for taking this on. I think there are a few things we need to do differently:

**Comment by ojeytonwilliams** [comment]
Hey @SadmanSyfe, don't feel like you have to delete the branch because git got a little messy.  It's definitely recoverable. I recommend reaching out in discord if you'd like some help!

**Comment by SadmanSyfe** [comment]
> Hey @SadmanSyfe, don't feel like you have to delete the branch because git got a little messy. It's definitely recoverable. I recommend reaching out in discord if you'd like some help!

@ojeytonwilliams Thank you for your advice. 😄 

> Hey @SadmanSyfe, thanks for taking this on. I think there are a few things we need to do differently:

I will create a new PR solving these issues.

**Comment by ojeytonwilliams** [comment]
Awesome, thanks @SadmanSyfe!

**Pull request closed without merging**